### PR TITLE
Fixed some serious lint errors

### PR DESCRIPTION
**Pull request by rjoleary**
Some of these might be bugs.

Signed-off-by: Ryan O'Leary <ryanoleary@google.com>

### CODE REVIEW

**Comment by rminnich** [comment]
can we take this a bit more slowly?  I don't understand the change to io at all, and I want to know why we've not seen this before -- is our circleci testing lacking?

File: pkg/boot/measurement.go [hunk 1]
```diff
@@ -69,6 +69,9 @@ func (mr *MeasuringReader) ReadRecord() (cpio.Record, error) {
 		switch rec.Name {
 		case "signature":
 			_, err = mr.signature.ReadFrom(uio.Reader(rec))
+			if err != nil {
```
Comment (rminnich) [inline pkg/boot/measurement.go]: if _, err := glab; err != nil {
...
}
?

Comment (rjoleary) [reply pkg/boot/measurement.go]: Ack, I'll fix this

File: cmds/io/io.go [hunk 2]
```diff
@@ -91,13 +91,13 @@ func main() {
 		}
 		switch t := i.val.(type) {
 		case uint8:
-			t = uint8(v)
+			b = uint8(v)
```
[the remark this answers is not in the document]
Comment (rjoleary) [reply cmds/io/io.go]: Line 94 in the original doesn't seem to have any effect. `t` is local to the switch statement and immediately goes out of scope on the next line.

File: pkg/boot/measurement.go (hunk 1 above)
Comment (hugelgupf) [reply pkg/boot/measurement.go]: No, this is currently deliberate. Please do not add this. I'll add a comment later.

Comment (rjoleary) [reply pkg/boot/measurement.go]: Sounds good. I'll remove this file and we can re-review it later.

Comment (rjoleary) [reply pkg/boot/measurement.go]: I'll make sure to add this linter to CI.

**Comment by rjoleary** [comment]
PTAL

**Comment by rminnich** [comment]
can we wrap this up?

**Comment by rjoleary** [comment]
We just need @hugelgupf to remove the X

File: xcmds/ash/main.go [hunk 3]
```diff
@@ -64,6 +65,11 @@ func main() {
 		log.Fatal(err)
 	}
 	r, err := t.Raw()
+	if err != nil {
+		// Error is non-fatal. The terminal might be missing some
+		// features, but can still be used.
+		log.Print(err)
```
Comment (hugelgupf) [inline xcmds/ash/main.go]: tell the user what this random error is about, e.g.

log.Printf("non-fatal terminal error: %v")

not sure if that's accurate

**Pull request merged**